library/ssl_tls13_client.c

```diff
@@ -1153,6 +1165,32 @@ int mbedtls_ssl_tls13_write_client_hello_exts( mbedtls_ssl_context *ssl,
     }
 #endif
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+    if( mbedtls_ssl_conf_tls13_some_psk_enabled( ssl ) &&
+        ( mbedtls_ssl_conf_has_static_psk( ssl->conf ) == 1
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
+          || ssl_tls13_early_data_ticket_verify( ssl )
+#endif
+        ) &&
+        ssl->conf->early_data_enabled == MBEDTLS_SSL_EARLY_DATA_ENABLED )
+    {
+        ret = mbedtls_ssl_tls13_write_early_data_ext( ssl, p, end, &ext_len );
+        if( ret != 0 )
+            return( ret );
+        p += ext_len;
+
+        ssl->handshake->early_data = MBEDTLS_SSL_EARLY_DATA_ON;
```

Comment:
https://github.com/Mbed-TLS/mbedtls/pull/6486/files#r1008293059 .

I am not sure if you forget push some codes.

Reply:
Actually, I think we need one field to indicate the status of early data, so I prefer to keep it.

Reply:
Please provide your opinion on https://github.com/Mbed-TLS/mbedtls/pull/6486/files#r1008293059 .  The comment is marked. But no code change and comments on it. I am confusing on it.

Reply:
Is it duplicated with `ssl->early_data_status`?

Reply:
Will add one new state for ssl->early_data_status to replace the field early_data.